tree/ntuple/v7/src/RPageStorageDaos.cxx

```diff
@@ -234,30 +253,62 @@ ROOT::Experimental::Detail::RPageSinkDaos::CommitSealedPageVImpl(std::span<RPage
       });
    locators.reserve(nPages);
 
+   uint32_t maxCageSz = fDaosContainer->GetDefaultCageSize();
+
    DescriptorId_t clusterId = fDescriptorBuilder.GetDescriptor().GetNClusters();
    std::size_t szPayload = 0;
+   std::size_t szCurrentCage;
+   uint32_t cageIdx;
+
+   std::vector<std::pair<unsigned, std::size_t>>
+      cageSizes; // Last page index included in cage and the cage's final size
 
    /// Aggregate batch of requests by object ID and distribution key, determined by the ntuple-DAOS mapping
    for (auto &range : ranges) {
+      szCurrentCage = 0;
+      cageIdx = fUseCaging ? fPageId.fetch_add(1) : fPageId.load();
+
       for (auto sealedPageIt = range.fFirst; sealedPageIt != range.fLast; ++sealedPageIt) {
+         
          const RPageStorage::RSealedPage &s = *sealedPageIt;
+
+         if (!fUseCaging || (szCurrentCage + s.fSize > maxCageSz)) {
+            cageSizes.emplace_back(locators.size() - 1, szCurrentCage);
+            szCurrentCage = 0;
+            cageIdx = fPageId.fetch_add(1);
+         }
+
          d_iov_t pageIov;
          d_iov_set(&pageIov, const_cast<void *>(s.fBuffer), s.fSize);
-         auto offsetData = fPageId.fetch_add(1);
 
-         RDaosKey daosKey = GetPageDaosKey<kDefaultDaosMapping>(clusterId, range.fColumnId, offsetData);
+         RDaosKey daosKey = GetPageDaosKey<kDefaultDaosMapping>(clusterId, range.fColumnId, cageIdx);
          auto odPair = RDaosContainer::ROidDkeyPair{daosKey.fOid, daosKey.fDkey};
          auto [it, ret] = writeRequests.emplace(odPair, RDaosContainer::RWOperation(odPair));
-         it->second.insert(daosKey.fAkey, pageIov);
+         it->second.Insert(daosKey.fAkey, pageIov);
 
          RNTupleLocator locator;
-         locator.fPosition = offsetData;
+         locator.fPosition = EncodeDaosPagePosition(cageIdx, szCurrentCage);
          locator.fBytesOnStorage = s.fSize;
          locators.push_back(locator);
 
+         szCurrentCage += s.fSize;
          szPayload += s.fSize;
       }
+
+      cageSizes.emplace_back(locators.size() - 1, szCurrentCage);
+   }
+
+   std::cout << "[LOC-V] Total number of page locators = " << locators.size() << std::endl;
```

Comment:
NOTE: ensure that (debug message) printing is not excessive here, to avoid overhead in the run time.